### DISCUSSION TITLE
Testing http service on EC2 provisioned by AutoScalingGroup(ASG)

### DISCUSSION THREAD

**Discussion by c0debreaker** [gruntwork-io/knowledge-base#736] [How do I]
I've been playing with terraform_http_example project today. It was quite cool!

In the project I am working on for the company, I'm using Terraform. I am using AWS AutoScaling Group to deploy ec2 instances. Tnat means, I use Terraform to deploy ASG.

If I want to test a web service on the EC2 instance, am I right that "terratest" will have to deploy ASG in order to deploy EC2 instances?

---

<ins datetime="2023-06-10T07:56:19Z">
  <p><a href="https://support.gruntwork.io/hc/requests/110251">Tracked in ticket #110251</a></p>
</ins>

**Comment by denis256** [comment]
Hi,
I would suggest having ASG configuration in Terraform code and Terratest execute "apply/destroy"
Having ASG deployment/destroy from Go code in Terratest will require a lot more code

**Reply by c0debreaker** [reply]
I'm confused with your response specifically on your second sentence.

I already have an existing ASG code in main.tf file. Does it mean I'm good? Or am I missing something?

**Reply by denis256** [reply]
Hello,
it is hard to say if something is missing without checking the code, but if already exists terraform files should be fine to apply changes through Terratest `terraform` module
